### ISSUE TITLE
system status: chromebook_asurada

### ISSUE THREAD

**Issue by hexdump0815** [hexdump0815/imagebuilder#60]
this issue is just a placeholder to potentially give feedback if my untested experimental images from https://github.com/hexdump0815/imagebuilder/releases/tag/220710-01 are working at all and in case they are working to note down what works and what not ...

**Comment by caljorden** [comment]
@hexdump0815 

Just a note, I successfully ran this release on my Asurada/Sipherion.  It booted right up to a full GUI.  Here is what I can tell you from the past hour or so of using it:

Works:
- Software rendering GUI (llvmpipe)
- Keyboard
- Trackpad
- WiFi
- Webcam
- Battery Meter
- Screen Brightness adjustments
- USB A port and USB-C ports
- KVM (Virtualization)

Not working:
 - Audio - kernel dmesg shows a bunch of ASoC errors ("Playback_1: ASoC: no backend DAIs enabled for Playback_1" repeats a large number of times with other messages).

Note: My system has 8GB of RAM, and all of it shows up properly.

Overall Summary: Very usable, very snappy system.  I would highly recommend it.

**Comment by hexdump0815** [comment]
@caljorden - thanks a lot for testing and reporting the info back ... regarding audio and some more info maybe useful asurada/spherion as well maybe have a look at https://github.com/hexdump0815/imagebuilder/issues/61 for inspiration

**Comment by hexdump0815** [comment]
@caljorden - i have built an experimental image with a v6.1 kernel which has working panfrost gpu support (but touchpad and sound does not work) - in case you want to give it a try: https://github.com/hexdump0815/imagebuilder/releases/tag/230109-01

**Comment by Pet0203** [comment]
> @caljorden - i have built an experimental image with a v6.1 kernel which has working panfrost gpu support (but touchpad and sound does not work) - in case you want to give it a try: https://github.com/hexdump0815/imagebuilder/releases/tag/230109-01

Can confirm GPU works. Would it be possible to fix touchpad and sound? I can try any images you want me to try. Would also be a nice learning opportunity for me!

**Comment by hexdump0815** [comment]
it might be worth experimenting with mainline linux on asurada meanwhile - see: https://oftc.irclog.whitequark.org/aarch64-laptops/2023-07-26#32341129;

**Comment by Pet0203** [comment]
@hexdump0815 would it be possible for you to release a new kernel/image for asurada with the latest stable kernel? There have been some fixes for the WiFi card stability on the mainline Linux tree that could potentially solve my biggest issue with Linux on Spherion. I've spent a few days trying to compile my own kernel but they refuse to boot. Thanks!

**Comment by hexdump0815** [comment]
@Pet0203 - i might look into that, but it will take a bit of time - i first plan to do something completely different and then i plan to have a look at a mainline kernel for corsola/asurada/cherry maybe in the range of october to december, maybe targetting v6.6 as a potential next lts kernel ... maybe at that time mainline will even be a bit easier to deal with on those devices

**Comment by Pet0203** [comment]
Can confirm mainline v6.7.0-rc1 is running on Spherion. My previous issues with the WiFi drivers timing out have been solved!
Audio no longer spams dmesg with errors, but there are still like 40 different sinks in the Audio Settings. ALSA detects the card correctly and loads the ucm config, but both pipewire and pulseaudio are getting the pcms instead of the "devices" defined in the ucm config.

Sleep is still not working, probably config error because the kernel has been patched for suspend/resume. Overall the system feels a lot more snappy and it feels like a gain on power efficiency, nothing I can confirm.

Last time I said that my kernels refused to boot. As it turned out, they didn't because I was able to SSH in. The issue was that the display output now runs on dp-aux as an edp panel.

**Comment by hexdump0815** [comment]
@Pet0203 - thanks a lot for the detailed report - this sounds similar to what was reported on irc a while ago: https://oftc.irclog.whitequark.org/aarch64-laptops/2023-07-26#32341129 and https://oftc.irclog.whitequark.org/linux-cros-arm/2023-09-04#32456692 ... do you use any extra patches and/or kernel config options besides pure mainline/defconfig? i guess you are aware, but just in case you are not: https://patchwork.kernel.org/project/linux-mediatek/list/ is a good place to watch out for potential further patches for mt8192 ... for rthe sound stuff i guess some more work on the ucm files might be required to get it working - first step is to get some audio driver to be probed successfully (i.e. something in /proc/asound/cards) which it seems you have already and then ucm config is required to make it work properly ...

i plan to try to get some kind of mainline working on corsola (mt8186) soon and will most probably run into similar problems there and that is not as far developed as asurada in mainline yet even as it seems - my first tries with the collabora chromeos mainline tree from https://gitlab.collabora.com/google/chromeos-kernel on corsola so far only resulted in a hanging kernel at some point into the boot with a few errors :)

**Comment by Pet0203** [comment]
Yeah I will give the UCM configs and puleaudio a second look at a later time. Really at this point the trackpad would be the most important thing for me to get working. For some reason the elan_i2c driver refuses to bind to the trackpad. I've dried using driverctl but to no avail. Really don't know what I should try next.

As for the config, I just used the defconfig for arm64 from [cros](https://chromium.googlesource.com/chromiumos/overlays/chromiumos-overlay/+/HEAD/eclass/cros-kernel/arm64_defconfig), excluded all platforms but Mediatek SoCs. Figured it was a good place to start when I didn't know what to include to get the display working. No patches though :)

**Comment by Pet0203** [comment]
Hi again! Thought I could mention that I fixed both the trackpad, speakers and also some Bluetooth issues while on Debian. Nowadays I've switched to Arch and suspend suddenly decided to start working aswell.

@hexdump0815 your repos have been an enormous help, thank you for your work!

**Comment by hexdump0815** [comment]
@Pet0203 - its nice to hear that you have it working better now ... could you maybe give some details what fixed the problems you mentioned, so that i can include ithe fixes into future images?

**Comment by Pet0203** [comment]
Sorry, I saw your message but I had lots of other stuff going on at that time.  Sadly I lost my solutions that fixed sound and bluetooth on debian when I switched to arch. Really gotten into this stuff but I'm still just essentially playing around. 

### Trackpad:
The fixes for the trackpad have been merged to mainline. It was just a matter of using hid-over-i2c instead of elan, because spherion has a synaptics trackpad not elan.

### Bluetooth:
Currently this is standing in my way, was way worse previously with a bunch of errors
 `[   10.978329] Bluetooth: hci0: Opcode 0x0c03 failed: -110`

### DP Alt Mode
I'm currently running Wens mtk-wip branch and have tried the DP alt mode. I could get a 640x480 image so that's something for sure but not there yet, or maybe I'm just missing something. Still getting a failed to switch mode error in dmesg which is weird.

### Suspend:
It wasn't fixed, it just didn't hang the system. Currently getting this through ectool: `Warning: Detected sleep hang! Waking host up!` So about 5 seconds after going to S3 the EC sends a wakeup. I think I used [this](https://www.kernel.org/doc/Documentation/ABI/testing/debugfs-cros-ec) to basically ignore it. Now it seems like the CPU maintains S3 at least? I just got ectool to work but still struggling with SuzyQ. If I remember correctly it worked with the chromeos kernel so I will try and find what made it work there.

### GPU
As for the GPU, Panfrost seems to be working fine. Mesa made some changes one day that borked my laptop. It was just pure coincidence that I installed arch when it worked, as neither the versions before or after it seemed to work. Haven't tried updating in a while though. 
<details>
  <summary>Can confirm that 24.2.5 works though!</summary>

```
OpenGL vendor string: Mesa
OpenGL renderer string: Mali-G57 (Panfrost)
OpenGL core profile version string: 3.1 Mesa 24.2.5 (git-3b9fcb7e4d)
OpenGL core profile shading language version string: 1.40
OpenGL core profile context flags: (none)
OpenGL core profile extensions:
OpenGL version string: 3.1 Mesa 24.2.5 (git-3b9fcb7e4d)
OpenGL shading language version string: 1.40
OpenGL context flags: (none)
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 3.1 Mesa 24.2.5 (git-3b9fcb7e4d)
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10
OpenGL ES profile extensions:
```
</details>

### FW-Devlink
Last year, Linux started to default to fw_devlink=rpm. My system didn't like that. I found that adding `driver_async_probe=* deferred_probe_timeout=2` to cmdline worked really well. I also set the config option `CONFIG_FW_DEVLINK_SYNC_STATE_TIMEOUT=y`, although I'm not sure I fully understand what it does. I'm not a driver dev. 

### Sound
The driver doesn't register the card through ALSA. Figuring this out has been a bit of a lower priority. Going to try different drivers.

**Comment by hexdump0815** [comment]
@Pet0203 - thanks a lot for the detailed feedback - re suspend: maybe s2idle can be used as a workaround? on modern hardware it can reach days of suspend sometimes ... but of course having real suspend would be better

re audio: did you have audio working properly at any point? (i.e. is it theoretically possible - would be better than nothing, even if it no longer works for you)

re wens wip-mtk tree: is that the one from the kernel git server?

**Comment by dinkumdingo** [comment]
@hexdump0815 I've tested all the images on my spherion and the trackpad only works on the first bullseye image. My trackpad shows up as `hid-over-i2c` on the first image, but doesn't show at all when calling `xinput list` on the newer experimental ones. Messing around with the `synaptics` and `libinput` packages didn't help.

Anyway, the audio works fine for me on the autumn test-image.
